### ISSUE TITLE
Windows Compatibility - Incorrect 3PG Prediction

### ISSUE THREAD

**Issue by jjczajka** [bdu91/group-contribution#2]
Hello,

I am trying to get the code running on a Windows System, but I am receiving a different prediction than the "examples.iypnb" notebook than for the case where a compound is not provided in the database. 
![image](https://user-images.githubusercontent.com/47155110/68488714-45f80780-020b-11ea-8b70-5aaab7e37331.png)

In the examples.iypnb notebook the output is: "array([-16974.39817563])".
![image](https://user-images.githubusercontent.com/47155110/68490042-e94a1c00-020d-11ea-99bc-60375de725e8.png)

I am not sure if this error may be due to the MarvinSuite installation or to some minor syntax changes (detailed below) that I made to get the code to run on Windows. After I download MarvinSuite and receive and install the License, I do not have a specific MarvinBean directory that is added to the MarvinSuite folder. However, I do have the "cxcalc" file in the "MarvinSuite\bin" folder. Have I installed the suite incorrectly? Will there be a different/updated "cxcalc" file that is installed within the "MarvinBean" folder?

The minor syntax changes I made where to the "chemaxon.py" file and are as follows (a picture of the code is below as well):

Line 22: 

    #orginial code 
    devnull = open('/dev/null', 'w')
    #new code
    devnull = open('NUL', 'w')

Line 26-27

    #orginial code 
    p1 = Popen(["echo", molstring], stdout=PIPE)
    p2 = Popen([CXCALC_BIN] + args, stdin=p1.stdout,
                   executable=CXCALC_BIN, stdout=PIPE, stderr=devnull)
    #new code
     os.chdir('D:\\ChemAxon\\MarvinSuite\\MarvinSuite\\bin')
     p1 = Popen(['echo', molstring], stdout=PIPE,shell=True)
     p2 = Popen([CXCALC_BIN] + args, stdin=p1.stdout, executable='cxcalc.exe', stdout=PIPE, stderr=devnull)

![image](https://user-images.githubusercontent.com/47155110/68489280-6a081880-020c-11ea-991e-8753c2cd460a.png)

Would any of these syntax changes lead to issues?

Thanks!

**Comment by bdu91** [comment]
Based on the large positive dGr calculated, it seems like dGf of 3pg is not calculated at all. It seems like there is some issue with calculating the pKas of 3pg. 
Can you run this in "examples.iypnb" notebook and let me know what you get?

`
import chemaxon; chemaxon.Get_pKas_Hs_zs_pH7smiles('O[C@H](COP(O)(O)=O)C(O)=O')
`

**Comment by jjczajka** [comment]
Running that code I get the following:

![image](https://user-images.githubusercontent.com/47155110/68681582-4228f580-0529-11ea-9b42-caa2875a2105.png)


I believe the pKas are correct, but I am not sure if there should also have been a fourth one output with a value around 13.4?

**Comment by bdu91** [comment]
The pKa looks correct, and dGf of 3pg should be calculated correctly. Currently I am waiting to get an updated chemAxon license so might take sometime to look into this. 

Meanwhile, after calculating the dGr for the reaction involving 3pg, can you run the following and show the output here? I will use those to compare with my output.

`print gc.to_calc_S_mat_T.shape`
`print gc.to_calc_ddGr`
`print gc.training_dG0_cc`
The last output will be a vector, can you put them into a txt file and drop it here?

We can start from these and narrow it down to where the issue might be.

**Comment by jjczajka** [comment]
Thanks for the help looking into this issue!

Here is the output:
![image](https://user-images.githubusercontent.com/47155110/68867064-dd9da000-06ba-11ea-80cc-85e5434443bd.png)

and the output from gc.training_dG0_cc is attached:

[output.txt](https://github.com/bdu91/group-contribution/files/3847052/output.txt)
[formatted_output.txt](https://github.com/bdu91/group-contribution/files/3847053/formatted_output.txt)

**Comment by bdu91** [comment]
Comparing your output with mine, it seems like the dGf of 3pg is not calculated at all (you can check this by `print gc.training_dG0_cc[-1]`, which will give you 0), there is no problem with chemaxon related calculations.

I am suspecting the compound group of 3pg is not correctly calculated. Can you print the output of this: 
`import numpy as np`
`np.nonzero(gc.TECRDB_compounds_data_dict['3pg_-3']['updated_groups'])`
It should give you "(array([ 41, 122, 130, 160]),)"

If not, I am suspecting the openbabel is not set up correctly in your environment.

**Comment by jjczajka** [comment]
Yes, it seems like that is the problem. The print gc.training_dG0_cc[-1] gave me a 0, while I did get the correct output from np.nonzero(gc.TECRDB_compounds_data_dict['3pg_-3']['updated_groups']). 

I had the openBabel3.0.0gui originally installed with the 2.4.1 openbabel bindings. I had to uninstall both installations as well as delete some openbabel folders from my directories that were not removed. After that, installing the openBabel2.4.1gui and bindings worked.

Thanks again for the help!
